src/bin/nydusd/daemon.rs

```diff
@@ -248,12 +258,27 @@ pub trait NydusDaemon: DaemonStateMachineSubscriber {
     fn get_state(&self) -> DaemonState;
     fn set_state(&self, s: DaemonState);
     fn trigger_exit(&self) -> DaemonResult<()> {
+        let s = self.get_state();
+
+        if s == DaemonState::STOPPED {
+            return Ok(());
+        }
+
+        if s == DaemonState::INIT {
+            return self.on_event(DaemonStateMachineInput::Stop);
+        }
+
+        if s == DaemonState::RUNNING {
+            self.on_event(DaemonStateMachineInput::Stop)?;
```

Comment:
Why not merge the two events handling if blocks.
They both react on `Stop` event

Reply:
The `INIT` status can transfer to `DIE` directly, while the `RUNNING` is transferred to `DIE` by `READY`. So After `stop` on `RUNNING`, there should be another `exit` event.

Reply:
> Why not merge the two events handling if blocks. They both react on `Stop` event

The `INIT` can transfer to `DIE` by `stop` directly. While the `RUNNING` is transferred to `DIE` via `READY`. For `INIT`, there is only one `stop` event, instead of two `stop` events for `RUNNING`.